src/Routers/AggregateRouter.js

```diff
@@ -11,20 +11,20 @@ export class AggregateRouter extends ClassesRouter {
     if (body.distinct) {
       options.distinct = String(body.distinct);
     }
-    if (body.hint) {
-      options.hint = body.hint;
+    if (Object.prototype.hasOwnProperty.call(body, 'hint')) {
```

Comment:
We can revert the changes and keep handleFile original logic.

A simple more efficent change would be adding a filter to `getPipeline` to avoid the unnecessary delete operation.
Also Object.keys won't be a problem when hint is present on the body prototype chain.

`static getPipeline(body) {
    let pipeline = body.pipeline || body;
    if (!Array.isArray(pipeline)) {
      pipeline = Object.keys(pipeline)
        .filter(key => pipeline[key] !== undefined)
        .map(key => {
          return { [key]: pipeline[key] };
        });
    }

  ....
  }`

Reply:
Sounds good, would you like to make that change?

Reply:
Sure.